wgsl/index.bs

```diff
@@ -8053,12 +7875,390 @@ of the following occur:
+
+    The number of bytes reserved in the struct for this member.
+
+    This number [=shader-creation error|must=] be at least the [=byte-size=] of the type of the member:
+    <p algorithm="byte-size constraint">
+    If `size(`|n|`)` is applied to a member with type |T|, then [=SizeOf=](|T|)&nbsp;&leq;&nbsp;|n|.
+    </p>
+
+    See [[#memory-layouts]]
+
+  <tr><td><dfn noexport dfn-for="attribute">`workgroup_size`</dfn>
+    <td>One, two or three parameters.
+
+    Each parameter [=shader-creation error|must=] be a [=const-expression=] or an [=override-expression=].
+    All parameters [=shader-creation error|must=] be the same type, either [=i32=] or [=u32=].
+
+    A [=shader-creation error=] results if any specified parameter is a
+    [=const-expression=] that evaluates to a non-positive value.
+
+    A [=pipeline-creation error=] results if any specified parameter evaluates
+    to a non-positive value or exceeds an upper bound specified by the WebGPU
+    API, or if the product of the parameter values exceeds the upper bound
+    specified by the WebGPU API (see [[WebGPU#limits]]).
+    <td>[=shader-creation error|Must=] be applied to a [=compute shader stage|compute shader=] entry point function.
+    [=shader-creation error|Must not=] be applied to any other object.
+
+    Specifies the x, y, and z dimensions of the [=workgroup grid=] for the compute shader.
+
+    The first parameter specifies the x dimension.
+    The second parameter, if provided, specifies the y dimension, otherwise is assumed to be 1.
+    The third parameter, if provided, specifies the z dimension, otherwise is assumed to be 1.
+
+</table>
+
+The <dfn noexport>shader stage attributes</dfn> below
+designate a function as an [=entry point=] for a particular [=shader stage=].
+These attributes [=shader-creation error|must=] only be applied to [=function declarations=],
+and at most one may be present on a given function.
+They take no parameters.
+
+<table class='data'>
+  <caption>Shader Stage Attributes</caption>
+  <thead>
+    <tr><th>Attribute<th>Description
+  </thead>
+
+  <tr><td><dfn noexport dfn-for="attribute">`vertex`</dfn><br>
+    <td>Declares the function to be an [=entry point=] for the [=vertex shader stage=]
+    of a [=GPURenderPipeline|render pipeline=].
+
+  <tr><td><dfn noexport dfn-for="attribute">`fragment`</dfn><br>
+    <td>Declares the function to be an [=entry point=] for the [=fragment shader stage=]
+    of a [=GPURenderPipeline|render pipeline=].
+
+  <tr><td><dfn noexport dfn-for="attribute">`compute`</dfn><br>
+    <td>Declares the function to be an [=entry point=] for the [=compute shader stage=]
+    of a [=GPUComputePipeline|compute pipeline=].
+
+</table>
+
+<pre class=include>
+path: syntax/attribute.syntax.bs.include
+</pre>
+<pre class=include>
+path: syntax/attrib_end.syntax.bs.include
+</pre>
+<pre class=include>
+path: syntax/diagnostic_control.syntax.bs.include
+</pre>
+
+# Directives # {#directives}
```

Comment:
This section feels odd in this position. It's very high level, but sandwiched between lower level things (attributes, and entry points).

Suggest moving it to after Textual Structure. That way it feels more like, "now that you know how to text is broken up into meaningful sentences, here's the first thing you may encounter, and it affects the whole program".

It also brings the "global diagnostic filter" closer to the local diagnostic filter bits too.